### PR TITLE
feat(Profile): add player stats and recent games to profile page

### PR DESCRIPTION
**Pull request by jacobnisnevich**
image

### CODE REVIEW

File: src/common/util/firebase.ts [hunk 1]
```diff
@@ -219,3 +225,33 @@ export function indexParsedSentence(sentence: string, tokens: string[], js: stri
...
 }
+
+export async function getRecentGamesByUserId(userId: string): Promise<Array<object>> {
+  const blueGames = await getRecentGamesForColorByUserId(userId, 'blue');
+  const orangeGames = await getRecentGamesForColorByUserId(userId, 'orange');
+
+  return Object.values({...blueGames, ...orangeGames});
+}
+
+async function getRecentGamesForColorByUserId(userId: string, color: string): Promise<object> {
```
Comment (AlexNisnevich) [inline src/common/util/firebase.ts]: as above, I think `object` can be `w.SavedGame`

File: src/common/util/firebase.ts [hunk 2]
```diff
@@ -219,3 +225,33 @@ export function indexParsedSentence(sentence: string, tokens: string[], js: stri
...
+  const snapshot = await fb.database()
+    .ref('games')
+    .orderByChild(`players/${color}`)
+    .equalTo(userId)
+    .once('value');
+  return snapshot.val();
+}
+
+export async function getCardsCreatedCountByUserId(userId: string): Promise<number> {
```
Comment (AlexNisnevich) [inline src/common/util/firebase.ts]: nit: maybe this method and the one below could somehow be deduplicated? I guess Rule of Three says don't bother until we have another similar method?

File: src/common/containers/Profile.js [hunk 3]
```diff
@@ -64,9 +73,49 @@ class Profile extends Component {
...
+  async loadProfileData() {
+    const userId = this.props.match.params.userId;
+    const recentGames = await getRecentGamesByUserId(userId);
+
+    this.loadRecentGamesData(userId, recentGames);
+    this.loadPlayerInfoData(userId, recentGames);
+  }
+
+  async loadRecentGamesData(userId, recentGames) {
+    const playerIds = uniq(recentGames.map((recentGame) => Object.values(recentGame.players)).flat());
```
Comment (AlexNisnevich) [inline src/common/containers/Profile.js]: There's no `flat()` method ...

File: src/common/components/profile/MatchmakingInfo.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    height: 'calc(100% - 35px)',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    fontSize: 36,
+    fontWeight: 100
+  }
+}; 
+
+// eslint-disable-next-line react/prefer-stateless-function
```
[the remark this answers is not in the document]
Comment (jacobnisnevich) [reply src/common/components/profile/MatchmakingInfo.js]: Can't do that with the `withStyles` I think?

File: src/common/containers/Profile.js [hunk 5]
```diff
@@ -64,9 +73,49 @@ class Profile extends Component {
...
+    const playerNamesList = await getUserNamesByIds(playerIds);
+    const playerNames = zipObject(playerIds, playerNamesList);
+
+    this.setState({ recentGames, playerNames });
+  }
+
+  async loadPlayerInfoData(userId, recentGames) {
```
[the remark this answers is not in the document]
Comment (jacobnisnevich) [reply src/common/containers/Profile.js]: Yup!

File: src/common/util/firebase.ts [hunk 6]
```diff
@@ -219,3 +225,33 @@ export function indexParsedSentence(sentence: string, tokens: string[], js: stri
     })
     .catch(noop);
 }
+
+export async function getRecentGamesByUserId(userId: string): Promise<Array<object>> {
```
[the remark this answers is not in the document]
Comment (jacobnisnevich) [reply src/common/util/firebase.ts]: Yup

File: src/common/util/firebase.ts [hunk 7]
```diff
@@ -118,6 +118,12 @@ export function saveUserData(key: string, value: any): void {
     .catch(noop);
 }
 
+export async function getUserNamesByIds(userIds: Array<string>): Promise<Array<string>> {
```
[the remark this answers is not in the document]
Comment (jacobnisnevich) [reply src/common/util/firebase.ts]: Oh good to know

File: src/common/util/firebase.ts (hunk 1 above)
Comment (jacobnisnevich) [reply src/common/util/firebase.ts]: Hmm close. Its a mapping of game IDs to SavedGames. So I think this is the most specific type?

```
Promise<{[gameId: string]: w.SavedGame}>
```

File: src/common/util/firebase.ts (hunk 2 above)
Comment (jacobnisnevich) [reply src/common/util/firebase.ts]: Hmm yeah I was thinking what a good name for the shared method would be

File: src/common/containers/Profile.js (hunk 3 above)
Comment (jacobnisnevich) [reply src/common/containers/Profile.js]: Will replace with _.flatten

File: src/common/components/Title.js [hunk 8]
```diff
@@ -1,21 +1,24 @@
...
+    opacity: 0.8,
+    borderTopRightRadius: 0,
+    borderBottomLeftRadius: 0
   }}>{text}</Paper>
 );
 
 Title.propTypes = {
-  text: string
+  text: string,
+  small: bool
```
[the remark this answers is not in the document]
Comment (jacobnisnevich) [reply src/common/components/Title.js]: Yeah it should, but not 100% strictly needed

**Pull request merged**